src/V3EmitCFunc.cpp

```diff
@@ -670,6 +670,8 @@ string EmitCFunc::emitVarResetRecurse(const AstVar* varp, const string& varNameP
                                                  depth + 1, suffix + "[" + ivar + "]");
         const string post = "}\n";
         return below.empty() ? "" : pre + below + post;
+    } else if (VN_IS(dtypep, StructDType) && !VN_AS(dtypep, StructDType)->packed()) {
+        return "";
```

Comment:
Complex types have default constructors, but what about primitives? We should probably reset them somehow. Either here or in unpacked struct constructors.

Reply:
I've added the resetting.

While I'd prefer using struct literals for that, it turned out necessary to "flatten" the field accesses here (example below)
```cpp
vlSelf->x__DOT__p.b.a = 0;
vlSelf->x__DOT__p.b.b = 0;
vlSelf->x__DOT__p.b.c = VL_RAND_RESET_I(4);
for (int __Vi1 = 0; __Vi1 < 4; ++__Vi1) {
    vlSelf->x__DOT__p.tab[__Vi1].a = 0;
    vlSelf->x__DOT__p.tab[__Vi1].b = 0;
    vlSelf->x__DOT__p.tab[__Vi1].c = VL_RAND_RESET_I(4);
}
```

The reason for that is that arrays inside structs are reset using for-statement, which cannot be placed inside a struct literal.

Emitting constructors would be a nice alternative, but the problem is that currently variable resetting code requires a pointer to the variable itself, which we cannot provide when we emit a type.